### ISSUE TITLE
Potential Groups selection works first time only

### ISSUE THREAD

**Issue by jamatrucola** [lsuits/block_quickmail#200]
Running Quickmail v1.5.5 (2015122912) on Moodle 3.0.  I'm finding that selecting an option in the "Potential Groups" list box only works once for each group.  After that, clicking on a group for a second time does nothing in the "Potential Recipients" box.

To replicate the problem, consider a course that has students categorized into 3 groups: A, B and C.  Load Quickmail.  By default, "All Groups" is selected under "Potential Groups" and no users are selected in "Potential Recipients".  Click Group A and watch all members of Group A get selected.  Click Group B and watch the members of Group A get deselected and the members of Group B get selected.  So far, so good.  Click Group A a second time.  The members of Group B get deselected but the members of Group A are not selected.  Click on Group A as many times as you like, but they'll never get selected again.  Click on Group C and watch its members get selected.  Click on any other group and watch the Group C members get deselected.  Now, you're done, since you've "used" each group once.  Click on any of the 3 groups and you'll never get users to be properly selected in the "Potential Recipients" box (unless you reload the page and start all over).

Note that in my example above, there's no user overlap between Groups A, B and C.  A little more testing showed that what's really a "one and done" usage is not the group, but the user.  Once a _user_ is selected and deselected via his membership in a "Potential Group", he'll never be able to be selected again via a Potential Group.  A way to see this very easily is: Load the Quickmail page, immediately choose "All Users" and watch every user get selected.  Then click on any other group.  Watch all your users get deselected and none of the desired users will be selected.  Now, that's it.  You're dead in the water.  You'll not be able to click on any other group that will highlight users.

**Comment by jamatrucola** [comment]
FYI, I modified js/selection.js as described #202 and #203 (changing attr to prop on lines 11 and 35) but the issue described here remains unresolved.

**Comment by markchaneyuk** [comment]
I would also like to confirm that we also experience this issue. Moodle 3.0.3 plugin  version: 2015122912

**Comment by jamatrucola** [comment]
Respectfully, I offer the same comment I wrote back on Jun 24.  I modified js/selection.js as described #202 and #203 (changing attr to prop on lines 11 and 35) but the issue described here remains unresolved.  Seeing that this issue was marked as Closed today, and that #202 was also marked closed, I looked more closely at the commit indicated there as the resolution.  It looks like the same change to selection.js described above, so I tested again, and again found that **this change does not resolve the bug described in this issue**.  I tested in Firefox and Chrome.  Are others finding that this code change does, in fact, resolve bug described in this issue?

**Comment by elliott2081** [comment]
I'll be testing it out and see if I can recreate your issue.

**Comment by nrbtushar** [comment]
we are using moodle3.0.4 and we have the same issue

**Comment by amandadoughty** [comment]
We have the issue in 3.0.6

**Comment by jamatrucola** [comment]
Quick update that may be helpful.  I wanted to play around with the new Moodle 3.2 software in a testing environment on my server, so I installed it as a completely separate instance of Moodle, right alongside my 3.1.2+ installation.  I have Quickmail v1.5.5 (2015122912) running on both installations, and **this issue is not present** on the Moodle 3.2 installation.  I should highlight that in the process of installing Moodle 3.2 I had to upgrade the version of PHP on the server (I went from PHP 5.4.45 to 5.6.29).  However I don't think this is significant since the Moodle 3.1.2+ installation is running on the same PHP 5.6.29 server alongside the new Moodle 3.2 installation ... and the issue still exists on the 3.1.2+ install but not the 3.2 install.
